### PR TITLE
Add parameter validation checks for length, allowed values, and number limits on defaultValue

### LINKED ISSUE #1228: defaultValue on constraints are not being evaluated
### Prerequisites

- [x] Write a descriptive title.
- [x] Make sure you are able to repro it on the latest version
- [x] Search the existing issues.

### Summary

Parameter constraints (`minValue`, `maxValue`, `minLength`, `maxLength`, `allowedValues`) are not being validated when parameters use their default values. Constraints are correctly validated when parameters are provided at runtime via input, but the validation is skipped entirely when a parameter falls back to its `defaultValue`.

### Steps to reproduce

1. Create a configuration with a parameter that has an invalid default value:

```yaml
$schema: https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/2024/04/config/document.json
parameters:
  port:
    type: int
    minValue: 1
    maxValue: 65535
    defaultValue: 99999  # Violates maxValue constraint
resources:
  - name: Echo Test
    type: Microsoft.DSC.Debug/Echo
    properties:
      output: "[parameters('port')]"
```

2. Run the configuration WITHOUT providing parameter input:

```powershell
dsc config get --file config.yaml
```

### Expected behavior

```console
DSC should validate the `defaultValue` (99999) against the `maxValue` constraint (65535) and raise an error:


ERROR Parameter input failure: Validation: Parameter 'port' has maximum value constraint of 65535 but is 99999
```

### Actual behavior

```console
The configuration runs successfully without any validation error, using the invalid default value of 99999.
```

### Error details

```console

```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.5.4
PSEdition                      Core
GitCommitId                    7.5.4
OS                             Microsoft Windows 10.0.26100
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Version

dsc 3.2.0-preview.7

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by Gijsreyn**
# PR Summary

This PR fixes a bug where parameter constraints (`minValue`, `maxValue`, `minLength`, `maxLength`, `allowedValues`) were not being validated when parameters used their `defaultValue`. Constraints were only validated for runtime input parameters, allowing invalid default values to pass through unchecked.

## PR Context

Fix #1228.

### CODE REVIEW

File: dsc/tests/dsc_parameters.tests.ps1 [hunk 1]
```diff
@@ -577,4 +577,164 @@ Describe 'Parameters tests' {
...
+"@
+
+        $testError = & {$config_yaml | dsc config get -f - 2>&1}
+        $LASTEXITCODE | Should -Be 4
+        $testError | Should -Match 'allowed values'
+    }
+
+    It 'Default values pass constraint validation for <type>' -TestCases @(
+        @{ type = 'string'; value = 'admin'; min = 3; max = 20 }
+        @{ type = 'int'; value = 8080; min = 1; max = 65535 }
```
Comment (SteveL-MSFT) [inline dsc/tests/dsc_parameters.tests.ps1:649]: Can you add tests for boundary conditions, exactly min and exactly max?

File: dsc/tests/dsc_parameters.tests.ps1 [hunk 2]
```diff
@@ -577,4 +577,164 @@ Describe 'Parameters tests' {
...
+                type: $type
+                minLength: $min
+                maxLength: $max
+                defaultValue: $value
+            resources:
+            - name: Echo
+              type: Microsoft.DSC.Debug/Echo
+              properties:
+                output: '[parameters(''param1'')]'
+"@
```
Comment (SteveL-MSFT) [inline dsc/tests/dsc_parameters.tests.ps1]: Since the two parts if the `if` statement are essentially the same, I think you should just simplify this to:

```
if ($type -eq 'string') {
  $value = "'$value'"
}
$config_yaml = ...
```

File: dsc/tests/dsc_parameters.tests.ps1 [hunk 3]
```diff
@@ -577,4 +577,164 @@ Describe 'Parameters tests' {
...
+        }
+
+        $out = $config_yaml | dsc config get -f - | ConvertFrom-Json
+        $LASTEXITCODE | Should -Be 0
+        if ($type -eq 'string') {
+            $out.results[0].result.actualState.output | Should -BeExactly 'admin'
```
Comment (SteveL-MSFT) [inline dsc/tests/dsc_parameters.tests.ps1]: Also here. I don't think you need to differentiate `string` here.  I believe `Should -BeEaxctly $value` would work.

File: dsc/tests/dsc_parameters.tests.ps1 [hunk 4]
```diff
@@ -577,4 +577,164 @@ Describe 'Parameters tests' {
...
+        } else {
+            $out.results[0].result.actualState.output | Should -BeExactly 8080
+        }
+    }
+
+    It 'Default values with allowedValues pass validation for <type>' -TestCases @(
+        @{ type = 'string'; value = 'dev'; allowed = @('dev', 'test', 'prod') }
+        @{ type = 'int'; value = 5; allowed = @(1, 5, 10) }
```
Comment (SteveL-MSFT) [inline dsc/tests/dsc_parameters.tests.ps1:681]: Maybe add one more test variation covering the last allowed value, doesn't matter if string or int.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Comment by Gijsreyn** [comment]
Thanks for the quick check @SteveL-MSFT. Resolved the remarks. Slightly rewritten the constraints tests.
